dummyserver/hypercornserver.py

```diff
@@ -25,28 +89,35 @@ async def graceful_shutdown(shutdown_event: threading.Event) -> None:
 
 
 async def _start_server(
-    config: hypercorn.Config,
+    config: Config,
     app: QuartTrio,
     ready_event: threading.Event,
     shutdown_event: threading.Event,
 ) -> None:
     async with trio.open_nursery() as nursery:
-        config.bind = await nursery.start(
-            functools.partial(
-                hypercorn.trio.serve,
-                app,
-                config,
-                shutdown_trigger=functools.partial(graceful_shutdown, shutdown_event),
+        try:
+            config.bind = await nursery.start(
+                functools.partial(
+                    hypercorn.trio.serve,
+                    app,
+                    config,
+                    shutdown_trigger=functools.partial(
+                        graceful_shutdown, shutdown_event
+                    ),
+                )
             )
-        )
-        ready_event.set()
+            ready_event.set()
+        except Exception:
+            print("Starting server failed", file=sys.stderr)
```

Comment:
Might be neater to use an anyio BlockingPortal using the https://anyio.readthedocs.io/en/stable/api.html#anyio.from_thread.BlockingPortal.start_task method, then startup failures should get propagated to the main thread properly

Reply:
Thanks, interesting suggestion! I'm not sure how to use that, though. (Also for what it's worth we use pure Trio, that no longer uses portal terminology.) Would you mind sharing more details/submitting a PR?

Reply:
@pquentin here's the PR https://github.com/urllib3/urllib3/pull/3508